test/alternator/test_streams.py

```diff
@@ -0,0 +1,373 @@
+                time.sleep(30)
+        except ClientError as ce:
+            print("ce {}".format(ce))
+            print("apa {}".format(ce.response['Error']['Code']))
+            if ce.response['Error']['Code'] == 'ResourceInUseException':
+                print("Stream(s) in use. Sleep 30s...")
+                time.sleep(30)
+                continue
+            raise
+            
+#
+# Cannot use fixtures. Because real dynamodb cannot _remove_ a stream
+# once created. It can only expire 24h later. So reusing test_table for 
+# example works great for scylla/local testing, but once we run against
+# actual aws instances, we get lingering streams, and worse: cannot 
+# create new ones to replace it, because we have overlapping types etc. 
+# 
+# So we have to create and delete a table per test. And not run this 
+# test to often against aws.  
+@contextmanager
+def create_stream_test_table(dynamodb, StreamViewType=None):
+    spec = { 'StreamEnabled': False }
+    if StreamViewType != None:
+        spec = {'StreamEnabled': True, 'StreamViewType': StreamViewType}
+    table = create_test_table(dynamodb, StreamSpecification=spec,
+        KeySchema=[ { 'AttributeName': 'p', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c', 'KeyType': 'RANGE' }
+        ],
+        AttributeDefinitions=[
+                    { 'AttributeName': 'p', 'AttributeType': 'S' },
+                    { 'AttributeName': 'c', 'AttributeType': 'S' },
+        ])
+    yield table
+    while True:
+        try:
+            table.delete()
+            return
+        except ClientError as ce:
+            if ce.response['Error']['Code'] == 'ResourceInUseException':
+                print('Could not delete table yet. Sleeping 5s.')
+                time.sleep(5)
+                continue;
+            raise
+
+def wait_for_active_stream(dynamodbstreams, table, timeout=60):
+    exp = time.process_time() + timeout
+    while time.process_time() < exp:
+        streams = dynamodbstreams.list_streams(TableName=table.name)
+        print(streams)
+        for stream in streams['Streams']:
+            desc = dynamodbstreams.describe_stream(StreamArn=stream['StreamArn'])['StreamDescription']
+            print(desc)
+            if not 'StreamStatus' in desc or desc.get('StreamStatus') == 'ENABLED':
+                arn = stream['StreamArn']
+                if arn != None:
+                    return arn;
+        print("Stream not available. Sleep 5s...")
+        time.sleep(5)
+    assert False
+
+# Local java dynamodb server version behaves differently from 
+# the "real" one. Most importantly, it does not verify a number of 
+# parameters, and consequently does not throw when called with borked 
+# args. This will try to check if we are in fact running against 
+# this test server, and if so, just raise the error here and be done 
+# with it. All this just so we can run through the tests on 
+# aws, scylla and local. 
```

Comment:
I found it useful to be able to use the local one. It does some compatibility checking, and more importantly allows one to figure out driver quirks before actually using real dynamo.

Reply:
Ok. I still think using the real Dynamo is easy enough as it is that the local one isn't necessary and worse - a distraction - because it caused you to try to support (in your code and tests) two DynamoDB implementations instead of one. But I don't strongly disagree that you leave it, since you already did the work.